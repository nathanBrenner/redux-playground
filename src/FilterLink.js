import React from "react";
import { store, Actions } from "./Store";

const FilterLink = ({ filter, children }) => {
  return (
    <button
      onClick={e => {
        e.preventDefault();
        store.dispatch({
          type: Actions.SET_VISIBILITY_FILTER,
          filter
        });
      }}
    >
      {children}
    </button>
  );
};

export default FilterLink;
