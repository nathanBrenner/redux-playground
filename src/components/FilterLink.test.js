import React from "react";
import FilterLink from "./FilterLink";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";

describe("FilterLink", () => {
  test("renders without crashing", () => {
    shallow(<FilterLink />);
  });

  test("snapshot", () => {
    const component = renderer.create(
      <MemoryRouter>
        <FilterLink filter="all" children="all" />
      </MemoryRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
