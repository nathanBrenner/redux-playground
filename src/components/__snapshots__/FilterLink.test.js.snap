// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterLink snapshot 1`] = `
<a
  aria-current="true"
  className="active"
  href="/"
  onClick={[Function]}
  style={
    Object {
      "color": "black",
      "textDecoration": "none",
    }
  }
>
  all
</a>
`;
